MediaBrowser.Controller/Entities/Folder.cs

```diff
@@ -644,12 +644,9 @@ public QueryResult<BaseItem> QueryRecursive(InternalItemsQuery query)
                 return PostFilterAndSort(items, query, true);
             }
 
-            if (!(this is UserRootFolder) && !(this is AggregateFolder))
+            if (!(this is UserRootFolder) && !(this is AggregateFolder) && query.ParentId.Equals(Guid.Empty))
```

Comment:
If C# allows `==` for `Guid`s, maybe do
```suggestion
            if (!(this is UserRootFolder) && !(this is AggregateFolder) && query.ParentId == Guid.Empty)
```